api/hasura/actions/createEpoch.ts

```diff
@@ -0,0 +1,118 @@
+import type { VercelRequest, VercelResponse } from '@vercel/node';
+import { DateTime } from 'luxon';
+import { z } from 'zod';
+
+import { authCircleAdminMiddleware } from '../../../api-lib/circleAdmin';
+import { EPOCH_REPEAT } from '../../../api-lib/constants';
+import { adminClient } from '../../../api-lib/gql/adminClient';
+import {
+  errorResponse,
+  errorResponseWithStatusCode,
+  zodParserErrorResponse,
+} from '../../../api-lib/HttpError';
+import {
+  createEpochInput,
+  composeHasuraActionRequestBody,
+} from '../../../src/lib/zod';
+
+async function handler(request: VercelRequest, response: VercelResponse) {
+  try {
+    const {
+      input: { payload: input },
+    } = composeHasuraActionRequestBody(createEpochInput).parse(request.body);
+    const { circle_id, repeat, start_date, days } = input;
+    let repeat_day_of_month = 0;
+    if (repeat > 0) {
+      const { epochs } = await adminClient.query({
+        epochs: [
+          {
+            limit: 1,
+            where: {
+              ended: { _eq: false },
+              circle_id: { _eq: circle_id },
+              repeat: { _gte: EPOCH_REPEAT.WEEKLY },
+            },
+          },
+          {
+            id: true,
+          },
+        ],
+      });
+      if (epochs.length) {
+        errorResponseWithStatusCode(
+          response,
+          { message: 'You cannot have more than one repeating active epoch.' },
```

Comment:
Consider adding the dates of the existing epoch too.
```suggestion
      if (epochs.length) {
      const overlappingEpoch = epochs.pop()
        errorResponseWithStatusCode(
          response,
          { message: `You cannot have more than one repeating active epoch. Dates overlap with epoch id #${overlappingEpoch.id}` },
```

Reply:
updated with start/end datetime